amethyst_animation/src/material.rs

```diff
@@ -0,0 +1,236 @@
+use fnv::FnvHashMap;
+
+use amethyst_assets::Handle;
+use amethyst_core::specs::Fetch;
+use amethyst_renderer::{Material, Texture, TextureOffset};
+use minterpolate::InterpolationPrimitive;
+
+use {AnimationSampling, ApplyData, BlendMethod};
+
+/// Textures used by texture animations
+#[derive(Debug, Default)]
+pub struct MaterialTextureSet {
+    textures: FnvHashMap<usize, Handle<Texture>>,
+    texture_inverse: FnvHashMap<Handle<Texture>, usize>,
+}
+
+impl MaterialTextureSet {
+    pub fn new() -> Self {
+        MaterialTextureSet {
+            textures: FnvHashMap::default(),
+            texture_inverse: FnvHashMap::default(),
+        }
+    }
+
+    pub fn handle(&self, index: usize) -> Option<Handle<Texture>> {
+        self.textures.get(&index).cloned()
+    }
+
+    pub fn index(&self, handle: &Handle<Texture>) -> Option<usize> {
+        self.texture_inverse.get(handle).cloned()
+    }
+
+    pub fn insert(&mut self, index: usize, handle: Handle<Texture>) {
+        self.textures.insert(index, handle.clone());
+        self.texture_inverse.insert(handle, index);
+    }
+
+    pub fn remove(&mut self, index: usize) {
+        if let Some(handle) = self.textures.remove(&index) {
+            self.texture_inverse.remove(&handle);
+        }
+    }
+
+    pub fn clear(&mut self) {
+        self.textures.clear();
+        self.texture_inverse.clear();
+    }
+}
+
+/// Sampler primitive for Material animations
+/// Note that material can only ever be animated with `Step`, or a panic will occur.
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+pub enum MaterialPrimitive {
+    Texture(usize),
+    Offset((f32, f32), (f32, f32)),
+}
+
+impl InterpolationPrimitive for MaterialPrimitive {
```

Comment:
Because of the trait bound on the interpolation functions in minterpolate that we use. I will implement these for some of the variants later on. For now we can only use step interpolation that won't actually call any of the functions

Reply:
:+1: